### PR TITLE
Host Picker logic

### PR DESCRIPTION
**Pull request by ZoroarkZark**
The main idea between this PR is to provide logic for the picker that will be used for the `Leave Event` feature,  in order for a current host to pick a new host for a particular event, using an view that would utilise this logic.

### CODE REVIEW

File: event-details-boundary/attendees-list/HostPicker.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            </View>
+          )}
+          attendeesList={attendeesList.attendeesList}
+          refresh={attendeesList.refresh}
+          isRefetching={attendeesList.isRefetching}
+        ></AttendeesListView>
+        <TouchableOpacity
+          style={style}
+          disabled={!selectedAttendeeId}
+          onPress={() => console.log("Continue")}
```
Comment (ZoroarkZark) [inline event-details-boundary/attendees-list/HostPicker.tsx:76]: submitted()

File: event-details-boundary/attendees-list/HostPicker.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+}
+
+export type PromoteToHostResult = "success" | "user-not-attending"
+
+export type UseEventHostPickerEnvironment = {
+  promoteToHost: (attendeeID: UserID) => Promise<PromoteToHostResult>
+  onSuccess: () => void
+}
+
+export const AttendeesListPicker = (
```
Comment (ZoroarkZark) [inline event-details-boundary/attendees-list/HostPicker.tsx:44]: EventHostPicker

File: event-details-boundary/attendees-list/HostPicker.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+) => {
+  const queryClient = useQueryClient()
+  const attendeesList = useAttendeesList(attendeesListProps)
+  const [selectedAttendeeId, setSelectedAttendeeId] = useState<
+    UserID | undefined
+  >()
+  const hostPickerMutation = useMutation(env.promoteToHost, {
+    onSuccess: (result, attendeeID) => {
+      if (result === "user-not-attending") {
+        setSelectedAttendeeId(undefined)
```
Comment (ZoroarkZark) [inline event-details-boundary/attendees-list/HostPicker.tsx:95]: Add updateAttendeesListQueryEvent but for removeAttendee

**Review by mhayes853 — APPROVED** [review]
TODO: `AttendeesListView` can handle loading/error states.

**Pull request merged**